lib/ecto/query/planner.ex

```diff
@@ -1105,16 +1105,47 @@ defmodule Ecto.Query.Planner do
     {{:in, in_meta, [left, right]}, acc}
   end
 
-  defp prewalk({:in, in_meta, [left, {:subquery, i}]}, kind, query, expr, acc, adapter) do
+  # Case when a subquery is at the Right Hand Side of either an in or quantifier based expression
+  defp prewalk({in_or_comp, op_meta, [left, {:subquery, i}]}, kind, query, expr, acc, adapter)
```

Comment:
I think you can reverse these changes too to what they were! (i.e. check only for `:in`)

Reply:
I'm also letting comparison operators go into this block for the same reasons as above. It's not necessarily to use a quantifier when comparing with a subquery. That's why I'm handling the case where a subquery without a quantifier is at the RHS of an "in" or comparison operator the same way.

Reply:
See the above. I missed you wanted to support row constructors too but I would like to skip the row constructors one for now. :)